doc/user_guide/data.rst

```diff
@@ -68,6 +68,21 @@ Similarly, we must also specify the data type when referencing data by URL:
         y='Miles_per_Gallon:Q'
     )
 
+You can use any URL visible to your frontend (i.e. JupyterLab, Jupyter Notebook, Colab, VSCode, Streamlit, etc.)
+
+All of those frontends have different mechanisms for accessing local files (if they have a mechanism at all), so there is no way to provide general advice for how to proceed. By way of example, consider a JupyterLab setup with seattle-weather.csv and a notebook in the top level folder. In this case, the full URL needs to be given.
```

Comment:
I think this phrasing makes sense in the context of a question about how to display local files, but sounds strange here when talking about URLs in general. Maybe start by saying something like "For accessing data by URL that is stored on your local machine, you can use any URL..."